hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineReportHandler.java

```diff
@@ -93,6 +93,9 @@ public void onMessage(PipelineReportFromDatanode pipelineReportFromDatanode,
       } catch(NotLeaderException ex) {
         // Avoid NotLeaderException logging which happens when processing
         // pipeline report on followers.
+      } catch(PipelineNotFoundException pnfe) {
```

Comment:
Does nt recon pipelineHandler processPipelineReport handled this already?


ReconPipelineReportHandler.java
```
    Pipeline pipeline = null;
    try {
      pipeline = reconPipelineManager.getPipeline(pipelineID);
    } catch (PipelineNotFoundException ex) {
      LOG.warn("Pipeline {} not found in Recon.", pipelineID);
      return;
    }
````

Reply:
what i want to do here is to handle this 
```
      Pipeline pipelineFromScm =
          scmServiceProvider.getPipeline(report.getPipelineID());
```
here , a scmclient is used to confirm the pipeline existence to scm, and this will take a grpc call.

Reply:
Do you mean hadoop rpc which uses failover proxy provider i believe

Reply:
yea, it will use , please take a look at [this](https://issues.apache.org/jira/secure/attachment/13027279/%E6%88%AA%E5%B1%8F2021-06-25%20%E4%B8%8B%E5%8D%884.18.35.png)